cli_tools/gce_ovf_import/ovf_utils/ovf_utils.go

```diff
@@ -35,42 +35,169 @@ const (
 	usbController          uint16 = 23
 )
 
-//Full list: https://vdc-download.vmware.com/vmwb-repository/dcr-public/da47f910-60ac-438b-8b9b-6122f4d14524/16b7274a-bf8b-4b4c-a05e-746f2aa93c8c/doc/vim.vm.GuestOsDescriptor.GuestOsIdentifier.html
+type osInfo struct {
+	description      string
+	importerOSIDs    []string
+	nonDeterministic bool
+}
+
+func (oi *osInfo) isDeterministic() bool {
+	return len(oi.importerOSIDs) == 1 && !oi.nonDeterministic
+}
 
-// Mapping OVF osType attribute to importer OS ID
-var ovfOSTypeToOSID = map[string]string{
-	"debian8Guest":          "debian-8",
-	"debian8_64Guest":       "debian-8",
-	"debian9Guest":          "debian-9",
-	"debian9_64Guest":       "debian-9",
-	"centos6Guest":          "centos-6",
-	"centos6_64Guest":       "centos-6",
-	"centos7Guest":          "centos-7",
-	"centos7_64Guest":       "centos-7",
-	"centos8Guest":          "centos-8",
-	"centos8_64Guest":       "centos-8",
-	"rhel6Guest":            "rhel-6",
-	"rhel6_64Guest":         "rhel-6",
-	"rhel7Guest":            "rhel-7",
-	"rhel7_64Guest":         "rhel-7",
-	"windows7Server64Guest": "windows-2008r2",
+func (oi *osInfo) hasImporterOSIDs() bool {
+	return len(oi.importerOSIDs) > 0
 }
 
-// Mapping potentially supported OVF osType values to possible importer OS ID values
+//Mapping OVF OS ID to OS info
+//Full list: http://schemas.dmtf.org/wbem/cim-html/2/CIM_OperatingSystem.html
+var ovfOSIDToImporterOSID = map[int16]osInfo{
```

Comment:
would you help me understand:
1.  what will happen if the image has a unlisted OS ID?
2. is there a way to leverage the os-detection project here?

Reply:
1. If user hasn't specified --os flag and there's no matching supported OS from OVF descriptor, error will be thrown here: https://github.com/GoogleCloudPlatform/compute-image-tools/blob/766e13993361ec666ffd6a9e9d17c67dd560a709/cli_tools/gce_ovf_import/ovf_importer/ovf_importer.go#L433
2. Yes, OS detection will be utilized for OVF import once available. We'll probably throw warnings or errors if what is detected and what is in the OVF descriptor don't match.